dds/src/dds/topic_definition/type_support.rs

```diff
@@ -154,7 +154,8 @@ const REPRESENTATION_OPTIONS: RepresentationOptions = [0x00, 0x00];
 
 /// This is a helper function to serialize a type implementing [`XTypesSerialize`] using the XTypes defined XCDR1 representation with LittleEndian endianness.
 pub fn serialize_rtps_xtypes_xcdr1_le(value: &impl XTypesSerialize) -> DdsResult<Vec<u8>> {
-    let mut writer = Vec::new();
+    let padded_length = (Xcdr1LeSerializer::bytes_len(value)? + 3) & !3;
+    let mut writer = Vec::with_capacity(padded_length + 4);
     writer.write_all(&CDR_LE)?;
     writer.write_all(&REPRESENTATION_OPTIONS)?;

```

Comment:
Weren't there some cases where these representation options should contain the padded length?

Reply:
here, it is because the padding is done at the end of this function. To prepare for it the pre-allocation length is also 'padded

Reply:
I don't find it now so it should be good :)